packages/perseus-editor/src/diffs/__tests__/item-diff.test.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+    testDependencies,
+    testDependenciesV2,
+} from "../../../../../testing/test-dependencies";
+import {registerAllWidgetsAndEditorsForTesting} from "../../util/register-all-widgets-and-editors-for-testing";
+import ItemDiff from "../item-diff";
+
+describe("ItemDiff", () => {
+    let unmockImageLoading: () => void;
+
+    beforeAll(() => {
+        registerAllWidgetsAndEditorsForTesting();
+    });
+
+    beforeEach(() => {
+        jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+            testDependencies,
+        );
+        unmockImageLoading = mockImageLoading();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+        unmockImageLoading();
+    });
+
+    it("renders an image widget in the diff view", () => {
+        // Arrange
+
+        const beforeItem: PerseusItem = generateTestPerseusItem({
+            question: generateTestPerseusRenderer({
+                content: "[[☃ image 1]]",
+                widgets: {
+                    "image 1": generateImageWidget({
+                        options: generateImageOptions({
+                            title: "Original Image",
+                            caption: "This is the original image",
+                            alt: "Original image description",
+                            backgroundImage: {
+                                url: "https://example.com/original-image.jpg",
+                                width: 400,
+                                height: 300,
+                            },
+                        }),
+                    }),
+                },
+            }),
+            answerArea: getDefaultAnswerArea(),
+            hints: [],
+        });
+
+        const afterItem: PerseusItem = generateTestPerseusItem({
+            question: generateTestPerseusRenderer({
+                content: "[[☃ image 1]]",
+                widgets: {
+                    "image 1": generateImageWidget({
+                        options: generateImageOptions({
+                            title: "Updated Image",
+                            caption: "This is the updated image",
+                            alt: "Updated image description",
+                            backgroundImage: {
+                                url: "https://example.com/updated-image.jpg",
+                                width: 500,
+                                height: 400,
+                            },
+                        }),
+                    }),
+                },
+            }),
+            answerArea: getDefaultAnswerArea(),
+            hints: [],
+        });
+
+        // Act
+        render(
+            <ItemDiff
+                before={beforeItem as any}
+                after={afterItem as any}
+                dependencies={testDependenciesV2}
+            />,
+        );
+
+        // Wait for the next tick to allow setTimeout to execute
+        act(() => {
+            jest.runAllTimers();
```

Comment:
Unfortunately we need this in order for the test to properly catch the regression. The diff has to render, and then the fake "onload" image event needs to fire and then get handled. I'm not sure if there's a better way to ensure these kinds of things fail in tests without running the jest timers.

Reply:
:/ yeah, there's places in Perseus that re-render that we have to handle like this. A necessary evil. Maybe one day we'll work Suspense through everything (although I'm not sure if that'd solve this problem).